swarm/api/http/server_test.go

```diff
@@ -892,3 +894,268 @@ func TestMethodsNotAllowed(t *testing.T) {
+	buffer, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer res.Body.Close()
+	body := string(buffer)
+
+	return res, body
+}
+
+func TestGet(t *testing.T) {
+	// Setup Swarm
+	srv := testutil.NewTestSwarmServer(t, serverFunc)
+	defer srv.Close()
+	
+	type GetTestCase struct{
+		uri string
+		method string
+		headers map[string]string
+		expectedStatusCode int
+		assertResponseBody string
+		verbose bool
+	}
+
+	testCases := []GetTestCase {
+		{
+			// Accept: text/html GET / -> 200 HTML, Swarm Landing Page
+			uri: fmt.Sprintf("%s/", srv.URL),
+			method: "GET",
+			headers: map[string]string{"Accept": "text/html"},
+			expectedStatusCode: 200,
+			assertResponseBody: "<a href=\"/bzz:/theswarm.eth\">Swarm</a>: Serverless Hosting Incentivised peer-to-peer Storage and Content Distribution",
+			verbose: false,
+		},
+		{
+			// Accept: application/json GET / -> 200 'Welcome to Swarm'
+			uri: fmt.Sprintf("%s/", srv.URL),
+			method: "GET",
+			headers: map[string]string{"Accept": "application/json"},
+			expectedStatusCode: 200,
+			assertResponseBody: "Welcome to Swarm!",
+			verbose: false,
+		},
+		{
+			// GET /robots.txt -> 200
+			uri: fmt.Sprintf("%s/robots.txt", srv.URL),
+			method: "GET",
+			headers: map[string]string{"Accept": "text/html"},
+			expectedStatusCode: 200,
+			assertResponseBody: "User-agent: *\nDisallow: /",
+			verbose: false,
+		},
+		{
+			// GET /path_that_doesnt exist -> 400
+			uri: fmt.Sprintf("%s/nonexistent_path", srv.URL),
+			method: "GET",
+			headers: map[string]string{},
+			expectedStatusCode: 400,
+			assertResponseBody: "",
+			verbose: false,
+		},
+		{
+			// GET bzz-invalid:/ -> 400
+			uri: fmt.Sprintf("%s/bzz:asdf/", srv.URL),
+			method: "GET",
+			headers: map[string]string{},
+			expectedStatusCode: 400,
+			assertResponseBody: "",
+			verbose: false,
+		},
+		{
+			// GET bzz-invalid:/ -> 400
+			uri: fmt.Sprintf("%s/tbz2/", srv.URL),
+			method: "GET",
+			headers: map[string]string{},
+			expectedStatusCode: 400,
+			assertResponseBody: "",
+			verbose: false,
+		},
+		{
+			// GET bzz-invalid:/ -> 400
+			uri: fmt.Sprintf("%s/bzz-rack:/", srv.URL),
+			method: "GET",
+			headers: map[string]string{},
+			expectedStatusCode: 400,
+			assertResponseBody: "",
+			verbose: false,
+		},
+		{
+			// GET bzz-invalid:/ -> 400
+			uri: fmt.Sprintf("%s/bzz-ls", srv.URL),
+			method: "GET",
+			headers: map[string]string{},
+			expectedStatusCode: 400,
+			assertResponseBody: "",
+			verbose: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run("GET " + testCase.uri, func(t *testing.T){
+			res, body := httpDo(testCase.method, testCase.uri, nil, testCase.headers, testCase.verbose, t)
+			if res.StatusCode != testCase.expectedStatusCode {
+				t.Fatalf("expected %s %s to return a %v but it didn't", testCase.method, testCase.uri, testCase.expectedStatusCode)
+			}
+			if testCase.assertResponseBody != "" && !strings.Contains(body, testCase.assertResponseBody) {
```

Comment:
when you omit the empty `assertResponseBody` from the test cases - the default value will be `""`, therefore the content won't be checked in this if (which is correct behaviour)